README.md

```diff
@@ -1,2 +1,137 @@
 # flatcar-mastodon
-Automation for deploying a Mastodon node on Flatcar
+Automation for deploying a Mastodon node on Flatcar.
+
+The files and config snippets in this repository allow to fully auto-provision a Mastodon node on Flatcar container linux.
+Set-up and configuration is automated via Ignition.
+Only instance-specific parameters need to be provided.
+
+## Features
+- Fully automated deployment based on very few settings (see HowTo below).
+- Automated HTTPS certificate retrieval and installation (via caddy + letsencrypt).
+- Includes elasticsearch for full-text indexing.
+
+### Planned features
+- Simple backup + restore.
+
+## HowTo
+
+Prerequisites: a domain name for your new Mastodon instance, and access to your domain registrar's DNS (see step 7.).
+
+1. Clone this repo.
+2. Edit [`caddy/Caddyfile`](caddy/Caddyfile) and set the domain name of your future instance, and the webmaster email address.
+3. Edit [`mastodon/admin.env`](mastodon/admin.env) and set the user name and email for Mastodon's initial Administrator account.
+   **DO NOT USE 'admin'**. This is a reserved name (Mastodon internally), using 'admin' will cause the initial set-up to fail.
+4. Edit [`mastodon/mastodon.env.template`](mastodon/mastodon.env.template) and set your future instance's domain (same as for Caddy above)
+   as well as an SMTP account for the instance to send notification emails to all users from.
+   Make sure the SMTP configuration is correct as this is the only way of using the initial admin account without manually logging
+   into the instance to retrieve the password.
+5. Generate Ignition deployment configuration based on your setting by running
+   ```shell
+   cat mastodon.yaml | docker run --rm -i -v $(pwd):/files ghcr.io/flatcar/ct:latest --files-dir /files > ignition.json
```

Comment:
Maybe we should switch to Butane?

Reply:
Hmmmm, not a bad idea though I only tested with ct right now. Will look into another test deployment this evening and will use butane-generated Ignition for that.

Reply:
I tried using butane following the official docs (https://docs.fedoraproject.org/en-US/fedora-coreos/producing-ign/#_getting_butane) but quay.io wants me to log in first?

If we can't get around the requirement of having a quay.io account I'm not going to use this because it will require more effort from our users who just want to deploy mastodon...

Reply:
Hmm, `docker pull quay.io/coreos/butane:release` worked just fine for me… Unless I misunderstood your problem.